pkg/apply/applier.go

```diff
@@ -114,192 +113,139 @@ func (a *Applier) prepareObjects(localInv inventory.InventoryInfo, localObjs obj
 	return localObjs, pruneObjs, nil
 }
 
-// Run performs the Apply step. This happens asynchronously with updates
-// on progress and any errors are reported back on the event channel.
+// Run performs the Apply step. This synchronously with updates
+// on progress and any errors reported back via callbacks.
```

Comment:
Run is a blocking method now. If caller needs concurrency they can always launch it in a goroutine. A typical caller doesn't need concurrency as it waits for the operation to be done. See tests for an example of a concurrent invocation.